### PR TITLE
IGListAdapter support for scrollViewDidEndDecelerating

### PR DESCRIPTION
**Pull request by plarson**
References #842

We use -[UIScrollViewDelegate scrollViewDidEndDecelerating:] delegate callback in the app we are building to hide/show elements when motion has ended.

### CODE REVIEW

File: Source/IGListScrollDelegate.h [hunk 1]
```diff
@@ -46,6 +46,14 @@ NS_SWIFT_NAME(ListScrollDelegate)
  */
 - (void)listAdapter:(IGListAdapter *)listAdapter didEndDraggingSectionController:(IGListSectionController *)sectionController willDecelerate:(BOOL)decelerate;
 
+/**
+ Tells the delegate that the section controller did end decelerating on screen.
+
+ @param listAdapter The list adapter whose collection view ended decelerating.
+ @param sectionController The visible section controller that ended decelerating.
+ */
+- (void)listAdapter:(IGListAdapter *)listAdapter didEndDeceleratingSectionController:(IGListSectionController *)sectionController;
```
Comment (jessesquires) [inline Source/IGListScrollDelegate.h]: breaking change here.

**Comment by plarson** [comment]
Ok @jessesquires, I made that method optional and added respondsToSelector checks

**Comment by rnystrom** [comment]
Yup lets make this optional with a note that it will be required in 4.0

**Comment by rnystrom** [comment]
I'm gonna bump this to 3.2 since its a new feature anyways. @plarson lmk if you're still able to work on this, if not we can take over the PR. Thanks!

**Comment by plarson** [comment]
Ok, I added a test. I also add a NS_SWIFT_NAME for the selector because it wasn't translating automatically to match the others.

**Comment by rnystrom** [comment]
@plarson awesome! We just need another test to cover IGListAdapter forwarding to other SCs (see example I posted above) and we're set!

**Comment by plarson** [comment]
My bad, added

File: Tests/IGListStackSectionControllerTests.m [hunk 2]
```diff
@@ -693,6 +693,34 @@ - (void)test_whenForwardingDidEndDraggingEvent_thatChildSectionControllersReceiv
...
+    [[mockScrollDelegate expect] listAdapter:self.adapter didEndDeceleratingSectionController:stack0.sectionControllers[0]];
+    [[mockScrollDelegate expect] listAdapter:self.adapter didEndDeceleratingSectionController:stack0.sectionControllers[1]];
+    [[mockScrollDelegate expect] listAdapter:self.adapter didEndDeceleratingSectionController:stack0.sectionControllers[2]];
+    [[mockScrollDelegate expect] listAdapter:self.adapter didEndDeceleratingSectionController:stack1.sectionControllers[0]];
+    [[mockScrollDelegate expect] listAdapter:self.adapter didEndDeceleratingSectionController:stack1.sectionControllers[1]];
+
+    [self.adapter scrollViewDidEndDecelerating:self.collectionView];
+
+    [mockScrollDelegate verify];
+}
```
Comment (rnystrom) [inline Tests/IGListStackSectionControllerTests.m]: Note that this test doesn't actually pass. The mock fails for `respondsToSelector` so these never get called, and expectations fail. Working on a fix atm.

**Pull request closed without merging**